### DISCUSSION TITLE
Should the server have GPU?

### DISCUSSION THREAD

**Discussion by FeryET** [Project-MONAI/MONAILabel#761] [Q&A] [answered]
Hi.

I'm looking for an annotation tool for our lab, and this looks really promising. The problem is that dedicating a gpu to this tool is not that feasible for us. Since we only use this for labelling images (I don't know about other functionalities of this tool), is having a GPU on the server necessary? Or it only requires the client to have a gpu? I'm a bit puzzled by the documentation.

**Comment by diazandr3s** [comment, marked as answer]
Hi @FeryET, thanks for opening this discussion. This is a good question.

We recommend having a GPU on the server-side only. This means where MONAI Label is installed. 

The client-side doesn't need to have a GPU.

If having a dedicated GPU for the MONAI Label server isn't possible, you could use MONAI Label for inference only (using CPU) and, in combination with the 3D Slicer manual tools, you could create labels that are later used to train/re-train the model in a GPU based PC.

Hope this helps. Please let us know.

**Reply by FeryET** [reply]
> Hi @FeryET, thanks for opening this discussion. This is a good question.
> 
> We recommend having a GPU on the server-side only. This means where MONAI Label is installed.
> 
> The client-side doesn't need to have a GPU.
> 
> If having a dedicated GPU for the MONAI Label server isn't possible, you could use MONAI Label for inference only (using CPU) and, in combination with the 3D Slicer manual tools, you could create labels that are later used to train/re-train the model in a GPU based PC.
> 
> Hope this helps. Please let us know.

Hello and thank you for your answer.

We want to use MONAI as an annotation only tool, without using any underlying models. Right now we don't have any models available for our tasks and we want to collect data. Is this possible with monai?

**Reply by diazandr3s** [reply]
Thanks for the update @FeryET.
Which image modality are you working on? What are you trying to segment? What is the approximate number of images you're segmenting?
MONAI Label allows you to quickly create pretrained models.
Please let us know.

**Reply by FeryET** [reply]
We're working on multiple modalities. Currently CT-Scan of brain images. We also don't have any gpu available for our labelling pipeline, and want to do both segmentation and classification labelling tasks as soon as possible. I thought maybe disabling the modelling pipeline and only using MONAILabel as an annotation tool would be faster for our usecase. Later on we will definitely add our models to the pipeline too.

**Reply by diazandr3s** [reply]
Thanks, @FeryET.
You could use MONAI Label without GPU, but that will significantly slow both training and inference. I guess you want to use MONAI Label for inference tasks only. Right? If yes, another limitation is the limited number of pretrained models currently available. Discarding the training option of MONAI Label will reduce the power of the framework itself. The available active learning strategies use the training or updating process to get better and faster segmentations.

**Reply by FeryET** [reply]
> Thanks, @FeryET. You could use MONAI Label without GPU, but that will significantly slow both training and inference. I guess you want to use MONAI Label for inference tasks only. Right? If yes, another limitation is the limited number of pretrained models currently available. Discarding the training option of MONAI Label will reduce the power of the framework itself. The available active learning strategies use the training or updating process to get better and faster segmentations.

I want to use MONAI to have an annotation pipeline, having models in the loop for retraining or providing inference is not a priority currently.

So this is the likely scenario:

1. We collect N annotated samples without any model in the loop. <--------------------------------  We are here
2. We train our baseline model later (which is a part of our research).
3. We add our own model to MONAI hopefully one day.
4. We use the active learning feature of MONAI on our tool in the future.

**Reply by diazandr3s** [reply]
Superb!

**Reply by bxin3444** [reply]
Hi @diazandr3s , I think I have a similar scenario to FeryET. May I please ask if there is an argument that we can use to switch between CPU and GPU mode? Currently, I have to manually remove a bit of initialization code to get the server started without GPU. 

Cheers,
Bowen

**Comment by SachidanandAlle** [comment]
https://docs.monai.io/projects/label/en/latest/installation.html#prerequisites
